lib/rules/func-name-matching.js

```diff
@@ -89,9 +92,22 @@ module.exports = {
     create(context) {
         const options = (typeof context.options[0] === "object" ? context.options[0] : context.options[1]) || {};
         const nameMatches = typeof context.options[0] === "string" ? context.options[0] : "always";
+        const considerPropertyDescriptor = options.considerPropertyDescriptor;
         const includeModuleExports = options.includeCommonJSModuleExports;
         const ecmaVersion = context.parserOptions && context.parserOptions.ecmaVersion ? context.parserOptions.ecmaVersion : 5;
 
+        /**
+         * Check wether node is a CallExpression on Object.
```

Comment:
Nit: s/wether/whether

Reply:
That's ok Kev. OS isn't easy.

* rebased on master
* fixed wether
* added test (and fixed TypeError)

```js
        {
            code: "({ value: function value() {} })",
            options: [{ considerPropertyDescriptor: true }]
        }
```

I couldn't get it to crash on `node.parent.parent.parent.parent`; is ASTNode safe-guarded against this?

Reply:
> is ASTNode safe-guarded against this?

I don’t think so. It looks like the `FunctionExpression` has four parents when an object literal is placed at the root of a program: `FunctionExpression -> Property -> ObjectExpression -> ExpressionStatement -> Program -> null`